src/main.cpp

```diff
@@ -242,29 +245,57 @@ Napi::Value DDWAFContext::run(const Napi::CallbackInfo& info) {
     Napi::Error::New(env, "Calling run on a disposed context").ThrowAsJavaScriptException();
     return env.Null();
   }
-  if (info.Length() < 2) {  // inputs, timeout
-    Napi::Error::New(env, "Wrong number of arguments, expected 2").ThrowAsJavaScriptException();
+
+  if (info.Length() < 2) {  // payload, timeout
+    Napi::Error::New(env, "Wrong number of arguments, 2 expected").ThrowAsJavaScriptException();
     return env.Null();
   }
+
   if (!info[0].IsObject()) {
-    Napi::TypeError::New(env, "First argument must be an object").ThrowAsJavaScriptException();
+    Napi::TypeError::New(
+            env,
+            "Payload data must be an object")
+        .ThrowAsJavaScriptException();
     return env.Null();
   }
+
+  Napi::Object payload = info[0].As<Napi::Object>();
+  Napi::Value persistent = payload.Get("persistent");
+  Napi::Value ephemeral = payload.Get("ephemeral");
+
+  if (!persistent.IsObject() && !ephemeral.IsObject()) {
+    Napi::TypeError::New(env, "Persistent or ephemeral must be an object").ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
   if (!info[1].IsNumber()) {
-    Napi::TypeError::New(env, "Second argument must be a number").ThrowAsJavaScriptException();
+    Napi::TypeError::New(env, "Timeout argument must be a number").ThrowAsJavaScriptException();
     return env.Null();
   }
+
   int64_t timeout = info[1].ToNumber().Int64Value();
   if (timeout <= 0) {
-    Napi::TypeError::New(env, "Second argument must be greater than 0").ThrowAsJavaScriptException();
+    Napi::TypeError::New(env, "Timeout argument must be greater than 0").ThrowAsJavaScriptException();
     return env.Null();
   }
 
+  ddwaf_object *ddwafPersistent = nullptr;
+
+  if (persistent.IsObject()) {
+    ddwafPersistent = static_cast<ddwaf_object *>(alloca(sizeof(ddwaf_object)));
```

Comment:
Why the allocation? in other cases it's not needed, no?

Reply:
That allocation occurs in the stack. `alloca` function allows you to allocate space in the stack conditionally so if the space is not needed the stack won't grow. Regarding the space allocated, it's automatically deallocated when the function ends due to the stack frame is freed.